pkg/controller/risingwave_controller.go

```diff
@@ -115,11 +120,56 @@ func (c *RisingWaveController) managerOpts(risingwaveMgr *object.RisingWaveManag
 	return opts
 }
 
-func (c *RisingWaveController) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
-	logger := log.FromContext(ctx)
+func (c *RisingWaveController) beforeReconcile(target types.NamespacedName, gvk schema.GroupVersionKind) {
+	metrics.IncControllerReconcileCount(target, gvk)
+}
+
+func (c *RisingWaveController) afterReconcile(
+	res *reconcile.Result,
+	err *error,
+	request reconcile.Request,
+	gvk schema.GroupVersionKind,
+	ctx context.Context,
+	reconcileStartTS time.Time) reconcile.Result {
```

Comment:
I think we don't need the return value here anymore since it is not used. Shall we remove it?

Reply:
True. We have to change the `res *reconcile.Request` though. Let me fix that.